pkg/ctl/cmdutils/configfile.go

```diff
@@ -272,7 +272,9 @@ func NewCreateClusterLoader(cmd *Cmd, ngFilter *filter.NodeGroupFilter, ng *api.
 				}
 			}
 		}
-
+		if clusterConfig.Karpenter != nil {
+			clusterConfig.SetDefaultKarpenterCreateServiceAccount()
+		}
```

Comment:
I would move this to `SetClusterConfigDefaults` which would allow you to perform defaulting in your tests without having to read a ClusterConfig file.

Reply:
The default is having `Karpenter` as nil.

Reply:
You can still add the same logic in `SetClusterConfigDefaults`.

Reply:
Alright, works for me